### ISSUE TITLE
[Bug]: ControlNet and Inpaint Mask not working for some preprocessors in A1111

### ISSUE THREAD

**Issue by abushyeye** [Mikubill/sd-webui-controlnet#2148]
### Is there an existing issue for this?

- [X] I have searched the existing issues and checked the recent builds/commits of both this extension and the webui

### What happened?

Using Automatic1111 and latest ControlNet, some of the preprocessors like LineArt, Canny, Softedge are not working. This happens specifically when "Only Masked" is selected. In some cases, the controlnet output is a black box, black box with slight horizontal lines, or a cropped out version of the input picture. 
![one (2)](https://github.com/Mikubill/sd-webui-controlnet/assets/146224435/7a75fc95-ac24-4766-970c-1123189751b5)
![two (1)](https://github.com/Mikubill/sd-webui-controlnet/assets/146224435/92e7c39e-1440-4dc6-9ccd-0d20b7733e5b)
![three (1)](https://github.com/Mikubill/sd-webui-controlnet/assets/146224435/ca3ee5d8-0623-4637-a669-4154356d201f)

Steps: 20, Sampler: Euler a, CFG scale: 3, Seed: 999, Size: 512x512, Model hash: 4c86efd062, Model: SDv1-5, VAE hash: c6a580b13a, VAE: vae-ft-mse-840000-ema-pruned.vae.pt, Denoising strength: 1, Init image hash: 26a8e14cb29034d412d74767b77559e6, Mask blur: 4, ControlNet 0: "Module: lineart_standard (from white bg & black line), Model: control_v11p_sd15_lineart [43d4be0d], Weight: 1, Resize Mode: Resize and Fill, Low Vram: False, Processor Res: 512, Guidance Start: 0, Guidance End: 1, Pixel Perfect: False, Control Mode: Balanced", Version: v1.6.0

EDIT: The SD Model used doesn't impact anything. I've tried the base 1.5, the 1.5 inpainting, and a bunch of other checkpoints with no impact on the CN output. Just to add. 

### Steps to reproduce the problem

1. Go to A1111 web ui
2. Go to img2Img tab
3. Go to inpaint upload tab
4. Upload image and mask of your choice
5. Set Inpaint area as "Only Masked"
6. Open ControlNet dropdown and add a picture. 
7. Select One of these control types from the list (Canny, Depth, Softedge, LineArt). I have tried all these. 
8. Select "Resize and Fill" as a resize mode. 
9. Generate

### What should have happened?

ControlNet should output properly sized images after processing which then gets fed to the geneartion of the masked part of the image. 
It's not resizing and filling properly in this instance. In past experiences, it just outputs black square.

### Commit where the problem happens

webui: version: [v1.6.0](https://github.com/AUTOMATIC1111/stable-diffusion-webui/commit/5ef669de080814067961f28357256e8fe27544f4)
controlnet: tried using v1.1.410 1.1.408, 1.1.313, 1.1.312,1.1.237, 1.1.134,1.1.230,1.1.221, and 1.1.179 to mention a few.

### Command Line Arguments

```Shell
No
```

### List of enabled extensions

![extensions](https://github.com/Mikubill/sd-webui-controlnet/assets/146224435/91e05c71-7565-4fd4-89fb-10be775407f4)

### Console logs

```Shell
ControlNet - INFO - Loading model: control_v11p_sd15_lineart [43d4be0d]
 ControlNet - INFO - Loaded state_dict from [/workspace/sd/stable-diffusion-webui/extensions/sd-webui-controlnet/models/control_v11p_sd15_lineart.pth]
 ControlNet - INFO - controlnet_default_config
 ControlNet - INFO - ControlNet model control_v11p_sd15_lineart [43d4be0d] loaded.
 ControlNet - INFO - Loading preprocessor: lineart_standard
 ControlNet - INFO - preprocessor resolution = 512
 ControlNet - INFO - ControlNet Hooked - Time = 10.276776790618896
100%|███████████████████████████████████████████| 16/16 [00:02<00:00,  7.04it/s]
```

**Comment by FNSpd** [comment]
Same thing. IP-adapter is completely useless in inpaint because of it

**Comment by abushyeye** [comment]
It's using the masked area + padding as an anchor to size the CNet input image. Instead of the expected Resizing of the CNet input image to update the image ratio. Hopefully someone will look at this.

**Comment by stubkan** [comment]
Really hope someone fixes this soon - this is a pretty big bug since this is a feature that is advertised on the front page of Controlnet and it simply doesnt work.

It makes using controlnet with inpainting almost impossible  unless you get lucky or tediously tweak the image sizes until the preprocessor gets the right image size.

Can confirm abushyeyes theory - this bug appears to be as inpaint resizes the original image for itself to inpaint, and then controlnet images used for input dont match this new image size and end up using a wrongly cropped segment of the controlnet input image.

**Comment by mookiexl** [comment]
From here: https://github.com/vladmandic/automatic/issues/2568

Changing line 747 in controlnet.py to
`      input_image = [np.asarray(x)[:, :] for x in input_image]`
seems to fix both this and [#2285](https://github.com/Mikubill/sd-webui-controlnet/issues/2285)

(in SDNext at least, can't test A1111 because AMD)

**Comment by Gushousekai195** [comment]
I have this issue, too!

All I want for Christmas is for it to be fixed!

**Comment by lulmer** [comment]
Has anyone looked into this or need help to solve it ?

**Comment by 2blackbar** [comment]
No, the dev keeps closing it suggesting we do it the wrong way, but cnet reference model works similar way and does not have that issue - does not crop the small area.
There are models that need to scale instead of cropping input image to the size of masked image and this is one of those cases.
picking cop resize and resize fill does not help here but should help if it would be working properly.
Well he should at least make it optionable to use whole or cropped area in inpainting only masked mode.
